### PR TITLE
unit10Task2Commit

### PR DESCRIPTION
**Pull request by DenysStetsenko54321**

### CODE REVIEW

File: src/pages/Inventory.page.js [hunk 1]
```diff
@@ -29,4 +31,67 @@ export class InventoryPage extends BaseSwagLabPage {
...
+        const price = parseFloat((await this.inventoryItemPrice.nth(id).textContent()).replace('$', ''));
+        const description = await this.page.locator('.inventory_item_desc').nth(id).textContent();
+        return { name, price, description };
+    }
+
+    async getItemInfoByIndex(index) {
+        const name = await this.inventoryItemName.nth(index).textContent();
+        const price = parseFloat((await this.inventoryItemPrice.nth(index).textContent()).replace('$', ''));
+        const description = await this.page.locator('.inventory_item_desc').nth(index).textContent();
+        return { name, price, description };
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js:47]: I did not get the purpose of this method. this code is the same as we have in the previous method

File: src/tests/addingRandomItemsToCard.spec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    test('Adding random items to the cart', async ({ inventoryPage, page }) => {
+        const listOfItems = await inventoryPage.getItemsList();
+        expect(listOfItems.length).toBe(6);
+
+        const addedItems = await inventoryPage.addRandomItemsToCart();
+        expect(addedItems.length).toBeGreaterThan(0);
+
+        await inventoryPage.goToCart();
+
+        const shoppingCartPage = new ShopingCartPage(page);
```
Comment (AlinaZhykharieva) [inline src/tests/addingRandomItemsToCard.spec.js]: why we are not doing it in the fixtures?

File: src/pages/Inventory.page.js [hunk 3]
```diff
@@ -29,4 +31,67 @@ export class InventoryPage extends BaseSwagLabPage {
...
+        const items = await this.inventoryItems.allTextContents();
+        return items;
+    }
+
+    async addRandomItemsToCart() {
+        const addedItems = [];
+
+        const itemCount = await this.inventoryItems.count();
+
+        const count = Math.floor(Math.random() * itemCount) + 1;
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js]: maybe it would be better randomAmount?

**Review by AlinaZhykharieva — CHANGES_REQUESTED** [review]

File: src/pages/Inventory.page.js [hunk 4]
```diff
@@ -29,4 +31,67 @@ export class InventoryPage extends BaseSwagLabPage {
...
+                addedItems.push(itemInfo);
+            } catch (error) {
+                console.error(`Error ${index}: ${error.message}`);
+            }
+        }
+
+        return addedItems;
+    }
+
+    generateRandomIndexes(maxIndex, count) {
```
Comment (AlinaZhykharieva) [inline src/pages/Inventory.page.js]: and this method should be moved to utils. it is not related directly to the page

File: src/tests/addingRandomItemsToCard.spec.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        expect(addedItems.length).toBeGreaterThan(0);
+
+        const shoppingCartPage = await inventoryPage.goToCartPage();
+        const cartItems = await shoppingCartPage.getCartItemsInfo();
+
+        for (const addedItem of addedItems) {
+            const match = cartItems.find((item) => item.name === addedItem.name
+                && item.price === addedItem.price
+                && item.description === addedItem.description);
+            expect(match).toBeDefined();
```
Comment (hlazkov) [inline src/tests/addingRandomItemsToCard.spec.js]: Could you please recheck on matchers, if there is any applicible? It should look like `toContain` or something similar.

File: src/tests/addingRandomItemsToCard.spec.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+test.describe('Unit 10', () => {
+    test.beforeEach(async ({ loginPage, inventoryPage }) => {
+        await loginPage.navigate();
+        await loginPage.performLogin(standardUser, standardPass);
+        await expect(inventoryPage.headerTitle).toBeVisible();
+    });
+
+    test('Adding random items to the cart', async ({ inventoryPage }) => {
+        const listOfItems = await inventoryPage.getItemsList();
+        expect(listOfItems.length).toBe(6);
```
Comment (hlazkov) [inline src/tests/addingRandomItemsToCard.spec.js]: This test will fail one day :) Do we really need to assert this?

**Review by hlazkov — CHANGES_REQUESTED** [review]
Looks nice! But please update the test itself!

**Review by AlinaZhykharieva — APPROVED** [review]

**Pull request merged**